apis/Google.Devtools.AspNet/Google.Devtools.AspNet/CloudErrorReporting/CloudErrorReportingExceptionLogger.cs

```diff
@@ -94,15 +100,32 @@ public sealed class CloudErrorReportingExceptionLogger : ExceptionLogger
         /// <inheritdoc />
         public override Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
         {
-            ReportedErrorEvent errorEvent = CreateReportRequest(context);
-            return _client.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            if (_clientTask.IsCompleted)
+            {
+                ReportedErrorEvent errorEvent = CreateReportRequest(context);
+                return _clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            }
+
+            if (Interlocked.Increment(ref _taskCounter) <= MaxWaitingTasks)
+            {
+                return _clientTask.ContinueWith(clientTask =>
+                {
+                    if (clientTask.IsFaulted)
+                    {
+                        throw new InvalidOperationException("The client task has faulted.");
+                    }
+                    ReportedErrorEvent errorEvent = CreateReportRequest(context);
+                    return clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
```

Comment:
Ah I see, I didn't think about that.  That sounds reasonable but it looks like RunContinuationsAsynchronously is available in 4.6.  Is it worth changing?  I'm not sure what else I could do here